commons/phone/phone_relative_location.avsc

```diff
@@ -6,12 +6,17 @@
   "fields": [
     {"name": "time", "type": "double", "doc": "device timestamp in UTC (s)"},
     {"name": "timeReceived", "type": "double", "doc": "device receiver timestamp in UTC (s)"},
-    {"name": "provider", "type": {"name": "LocationProvider", "type": "enum", "symbols": ["GPS", "NETWORK", "OTHER"]}, "doc": "Android provider of the location data"},
-    {"name": "latitude", "type": "double", "doc": "relative latitude from an initial random reference latitude (degrees with range [-90, 90])", "default": "NaN"},
-    {"name": "longitude", "type": "double", "doc": "relative longitude from an initial reference longitude (degrees with range [-180, 180])", "default": "NaN"},
-    {"name": "altitude", "type": "float", "doc": "height above the WGS 84 reference ellipsoid (m)", "default": "NaN"},
-    {"name": "accuracy", "type": "float", "doc": "The estimated horizontal accuracy of this location, radial (m). We define horizontal accuracy as the radius of 68% confidence. In other words, if you draw a circle centered at this location's latitude and longitude, and with a radius equal to the accuracy, then there is a 68% probability that the true location is inside the circle.", "default": "NaN"},
-    {"name": "speed", "type": "float", "doc": "speed over ground (m/s)", "default": "NaN"},
-    {"name": "bearing", "type": "float", "doc": "the horizontal direction of travel of this device (degrees with range (0, 360])", "default": "NaN"}
+    {"name": "provider", "type": {
+      "name": "LocationProvider",
+      "type": "enum",
+      "doc": "Type of provider of location data.",
+      "symbols": ["GPS", "NETWORK", "OTHER"]
+      }, "doc": "Android provider of the location data"},
+    {"name": "latitude", "type": ["null", "double"], "doc": "relative latitude from an initial random reference latitude (degrees with range [-90, 90])", "default": null},
+    {"name": "longitude", "type": ["null", "double"], "doc": "relative longitude from an initial reference longitude (degrees with range [-180, 180])", "default": null},
+    {"name": "altitude", "type": ["null", "float"], "doc": "height above the WGS 84 reference ellipsoid (m)", "default": null},
+    {"name": "accuracy", "type": ["null", "float"], "doc": "The estimated horizontal accuracy of this location, radial (m). We define horizontal accuracy as the radius of 68% confidence. In other words, if you draw a circle centered at this location's latitude and longitude, and with a radius equal to the accuracy, then there is a 68% probability that the true location is inside the circle.", "default": null},
+    {"name": "speed", "type": ["null", "float"], "doc": "speed over ground (m/s)", "default": null},
+    {"name": "bearing", "type": ["null", "float"], "doc": "the horizontal direction of travel of this device (degrees with range (0, 360])", "default": null}
```

Comment:
We may follow this approach on all other schemas: `time` and `timeReceived` cannot be null and do not have default, while all the other fields can be either `optional`  fields or enum with `UNKNOW` as default.

@blootsvoets What do you think?

Reply:
Yes, that would be my preferred approach for schemas with a lot of values that are may or may not be set. For some schemas, having `null` defaults in a field would add less value. It does slightly increase the processing needed (union traversal and intermediate objects) and additional checks at the analysis stage (remove/correct null values before actual analysis). Once a field has union-null type, a null check is always necessary.